src/Backups/BackupInfo.cpp

```diff
@@ -98,4 +98,29 @@ String BackupInfo::toStringForLogging() const
     return toAST()->formatForLogging();
 }
 
+void BackupInfo::copyS3CredentialsTo(BackupInfo & dest) const
+{
+    /// named_collection case, no need to update
+    if (!dest.id_arg.empty() || !id_arg.empty())
+        throw Exception(ErrorCodes::BAD_ARGUMENTS, "use_same_s3_credentials_for_base_backup is not compatible with named_collections");
+
+    if (backup_engine_name != "S3")
+        throw Exception(ErrorCodes::BAD_ARGUMENTS, "use_same_s3_credentials_for_base_backup supported only for S3, got {}", toStringForLogging());
+    if (dest.backup_engine_name != "S3")
+        throw Exception(ErrorCodes::BAD_ARGUMENTS, "use_same_s3_credentials_for_base_backup supported only for S3, got {}", dest.toStringForLogging());
+
+    /// copyS3CredentialsTo() should not be called in this case, since it
+    /// should fail in registerBackupEngineS3(), hence LOGICAL_ERROR
+    if ((args.size() != 1) && (args.size() != 3))
```

Comment:
Other checks are written in a way that BACKUP/RESTORE will fail if it can't copy credentials.
Perhaps for consistency with other checks it's better to check here

```
if (args.size() != 3)
```
instead of
```
if ((args.size() != 1) && (args.size() != 3))
```

Reply:
Ok, updated.